keyboards/skyloong/qk21/v1/info.json

```
@@ -0,0 +1,129 @@
+    },
+    "rgb_matrix":{
+        "driver": "WS2812",
+        "animations": {
+            "alphas_mods": false,
+            "band_pinwheel_sat": false,
+            "band_pinwheel_val": false,
+            "band_sat": false,
+            "band_spiral_sat": false,
+            "band_spiral_val": false,
+            "breathing": true,
+            "cycle_all": true,
+            "cycle_left_right": false,
+            "cycle_out_in": false,
+            "cycle_out_in_dual": false,
+            "cycle_pinwheel": true,
+            "cycle_spiral": true,
+            "cycle_up_down": true,
+            "dual_beacom": false,
+            "gradient_left_right": false,
+            "gradient_up_down": false,
+            "hue_breathing": false,
+            "hue_pendulum": false,
+            "hue_wave": false,
+            "jellybean_raindrops": true,
+            "pixel_fractal": false,
+            "pixel_rain": true,
+            "rainbow_moving_chevron":false,
+            "rainbow_pinwheels": true,
+            "raindrops": true,
+            "ranbow_beacon": true,
+            "typing_heatmap": true,
+            "solid_reactive_simple": true,
+            "solid_reactive_cross": true,
+            "splash": true
+        },
+        "layout": [
+            {"matrix": [5, 0], "x": 37, "y": 64, "flags": 4},
+            {"matrix": [5, 2], "x": 149, "y": 64, "flags": 4},
+            {"matrix": [5, 3], "x": 224, "y": 58, "flags": 4},
+
+            {"matrix": [4, 0], "x": 0, "y": 51, "flags": 4},
+            {"matrix": [4, 1], "x": 75, "y": 51, "flags": 4},
+            {"matrix": [4, 2], "x": 149, "y": 51, "flags": 4},
+
+            {"matrix": [3, 0], "x": 0, "y": 38, "flags": 4},
+            {"matrix": [3, 1], "x": 75, "y": 38, "flags": 4},
+            {"matrix": [3, 2], "x": 149, "y": 38, "flags": 4},
+            {"matrix": [3, 3], "x": 224, "y": 31, "flags": 4},
+
+            {"matrix": [2, 0], "x": 0, "y": 25, "flags": 4},
+            {"matrix": [2, 1], "x": 75, "y": 25, "flags": 4},
+            {"matrix": [2, 2], "x": 149, "y": 25, "flags": 4},
+
+            {"matrix": [1, 0], "x": 0, "y": 12, "flags": 8},
+            {"matrix": [1, 1], "x": 75, "y": 12, "flags": 4},
+            {"matrix": [1, 2], "x": 149, "y": 12, "flags": 4},
+            {"matrix": [1, 3], "x": 224, "y": 12, "flags": 4},
+
+            {"matrix": [0, 0], "x": 0, "y": 0, "flags": 4},
+            {"matrix": [0, 1], "x": 75, "y": 0, "flags": 4},
+            {"matrix": [0, 2], "x": 149, "y": 0, "flags": 4},
+            {"matrix": [0, 3], "x": 224, "y": 0, "flags": 4}
+
+        ],
+        "hue_steps": 20,
+        "max_brightness": 200,
+        "sat_steps": 20,
+        "speed_steps": 20,
+        "val_steps": 20
+    },
+    "url": "https://github.com/JZ-Skyloong",
+    "usb": {
+        "device_version": "1.0.0",
+        "pid": "0x6021",
+        "vid": "0x1EA7",
+        "max_power": 250
+    },
+    "default_layouts": ["numpad_6x4"],
```

"default_layout" isn't present in `data/schemas/keyboard.jsonschema`
```suggestion
    "community_layouts": ["numpad_6x4"],
```

---

I have checked the layout level, it seems that there is no 6x4 layout in the community, only 6x4 layout in default. What are your comments and suggestions? @dunk2k

---

> I have checked the layout level, it seems that there is no 6x4 layout in the community, only 6x4 layout in default. What are your comments and suggestions?

*Note: the following response is my personal understanding and not official QMK response*

`layout/default` serves as:
1. reference for determining if keyboard's layout matches a community layout
2. Container of keycodes for keymap default_<layoutName> when CI Build is compiling a keyboard that has community layout(s) defined.

`layouts/community` is where community members/contributors can input their keymaps and accompanying code for given community layouts.
For note: Community layouts can be used in conjunction with userspaces.

An analogy:
 Think of `layout/default` as templates and `layout/community` as drawings/forms completed using those templates.

---

@dunk2k Thank you for your insight. I will test the change to community layout to see if there is any problem. If everything is normal, I will adopt your suggestion.